### PR TITLE
Configure intermediate primary

### PR DESCRIPTION
**Pull request by kfyharukz**
Signed-off-by: Kazuhito MATSUDA <kazuhito.matuda@gmail.com>

### CODE REVIEW

File: accessor/mysql_status.go [hunk 1]
```diff
@@ -202,6 +216,12 @@ func GetMySQLClusterStatus(ctx context.Context, log logr.Logger, infra Infrastru
 
 func GetLatestInstance(ctx context.Context, db *sqlx.DB, status []MySQLInstanceStatus) (*int, error) {
 	var latest int
+	for i := 1; i < len(status); i++ {
```
Comment (umezawatakeshi) [inline accessor/mysql_status.go:219]: index should begin with 0 ?

File: controllers/mysql_clustering.go [hunk 2]
```diff
@@ -493,6 +503,10 @@ func waitForReplication(status *accessor.MySQLClusterStatus, cluster *mocov1alph
 }
 
 func acceptWriteRequest(status *accessor.MySQLClusterStatus, cluster *mocov1alpha1.MySQLCluster) []ops.Operator {
+	if status.IntermediatePrimaryOptions != nil {
```
Comment (zoetrope) [inline controllers/mysql_clustering.go:506]: Do you need to turn on `read-only` here?

File: accessor/mysql_status.go [hunk 3]
```diff
@@ -17,8 +17,16 @@ import (
...
-	Latest         *int
+	InstanceStatus             []MySQLInstanceStatus
+	Latest                     *int
+	IntermediatePrimaryOptions *IntermediatePrimaryOptions
+}
+
+type IntermediatePrimaryOptions struct {
+	MasterHost     string `db:"MasterHost"`
+	MasterUser     string `db:"MasterUser"`
+	MasterPassword string `db:"MasterPassword"`
```
Comment (zoetrope) [inline accessor/mysql_status.go:28]: In the future, I would like to be able to get these status through the API.
I don't welcome exposing a password.

File: accessor/mysql_status.go [hunk 4]
```diff
@@ -359,3 +379,46 @@ func GetMySQLCloneStateStatus(ctx context.Context, db *sqlx.DB) (*MySQLCloneStat
...
+
+	options, err := parseIntermediatePrimaryOptions(secret.Data)
+	return options, err
+}
+
+func parseIntermediatePrimaryOptions(options map[string][]byte) (*IntermediatePrimaryOptions, error) {
+	var result IntermediatePrimaryOptions
+	for k, v := range options {
+		switch k {
+		case "MASTER_HOST":
```
Comment (zoetrope) [inline accessor/mysql_status.go:402]: I prefer `PRIMARY` to `MASTER`.

File: operators/configure_intermediate_primary.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	_, err = db.Exec("SET GLOBAL rpl_semi_sync_slave_enabled=OFF")
+	if err != nil {
+		return err
+	}
+	_, err = db.Exec(`START SLAVE`)
+	return err
+}
+
+func (o configureIntermediatePrimaryOp) Describe() string {
+	return fmt.Sprintf("%#v", o)
```
Comment (zoetrope) [inline operators/configure_intermediate_primary.go:59]: The password will be outputted to log.

**Review by zoetrope — APPROVED** [review]
LGTM

**Pull request merged**